conjur/user/user_logic.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+
+"""
+UserLogic module
+
+This module is the business logic for handling all user-related activity
+"""
+
+# Builtins
+import logging
+import requests
+
+# Internals
+from conjur.credentials_from_file import CredentialsFromFile
+from conjur.errors import InvalidOperationException, OperationNotCompletedSuccessfullyException
+
+
+class UserLogic:
+    """
+    UserLogic
+
+    This class holds the business logic for handling user activity
+    """
+    def __init__(self, conjurrc_data, credentials_from_store, client, resource):
+        self.conjurrc_data=conjurrc_data
+        self.credentials_from_store=credentials_from_store
+        self.client=client
+        self.resource=resource
+
+    # pylint: disable=logging-fstring-interpolation,line-too-long
+    def rotate_api_key(self, resource_to_update):
+        """
+        Method to trigger two types of API key actions.
+        1. The user can either rotate their own API key or
+        2. Rotate another users API key
+        """
+        loaded_credentials = self.extract_credentials_from_credential_store()
+        user_from_credential_store = loaded_credentials['login_id']
```

Comment:
I would prefer a more concise name other than `user_from_credential_store`.
Maybe `current_user_name` or `logged_in_user_name`.
Adding the `name` suffix is important to denote its content, unless you will replace it with my suggestion for `User` class instance.